src/test/java/org/example/validators/JobApplicationValidatorTest.java

```diff
@@ -0,0 +1,103 @@
+package org.example.validators;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.SQLException;
+import org.example.daos.JobApplicationDao;
+import org.example.daos.JobRoleDao;
+import org.example.exceptions.AlreadyExistsException;
+import org.example.exceptions.DoesNotExistException;
+import org.example.exceptions.FileNeededException;
+import org.example.exceptions.FileTooBigException;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+public class JobApplicationValidatorTest {
+
+    private static final int MAX_FILE_SIZE_BYTES = 5 * 1024 * 1024; // 5MB in bytes
+
+    JobRoleDao jobRoleDao = mock(JobRoleDao.class);
+    JobApplicationDao jobApplicationDao = mock(JobApplicationDao.class);
+    JobApplicationValidator jobApplicationValidator = new JobApplicationValidator(jobRoleDao, jobApplicationDao);
+
+    @Nested
+    class ValidateAndProduceByteArrayTests {
```

Comment:
this is the first time in my career that I see nested class in use :D but.... do we actually need it?

Reply:
We dont need it. It makes tests more readable in case of more methotds in the future and assure tests independency in case of using @BeforeEach annotation (each test subclass can have different before each setup isolated from another setups)

Reply:
I would remove it tbh, does not improve visibility that much (and our project has no "future" :P YAGNI)